bundles/org.openhab.binding.surepetcare/src/main/java/org/openhab/binding/surepetcare/internal/handler/SurePetcareDeviceHandler.java

```diff
@@ -0,0 +1,254 @@
+        }
+
+    }
+
+    private void handleCurfewCommand(ChannelUID channelUID, Command command) {
+        String channelUIDBase = channelUID.getIdWithoutGroup().substring(0,
+                channelUID.getIdWithoutGroup().length() - 1);
+        int slot = Integer.parseInt(channelUID.getAsString().substring(channelUID.getAsString().length() - 1));
+
+        synchronized (petcareAPI) {
+            boolean requiresUpdate = false;
+            SurePetcareDevice device = petcareAPI.getDevice(thing.getUID().getId());
+            if (device != null) {
+                SurePetcareDeviceControl existingControl = device.control;
+                logger.debug("Old Curfew Control: {}", existingControl);
+
+                SurePetcareDeviceCurfew curfew = existingControl.curfewList.get(slot - 1);
+                logger.debug("channelUIDBase: {}", channelUIDBase);
+
+                switch (channelUIDBase) {
+                    case DEVICE_CHANNEL_CURFEW_ENABLED:
+                        if (command instanceof OnOffType) {
+                            logger.debug("old enables: {}, command: {}", curfew.enabled, command.equals(OnOffType.ON));
+                            if (curfew.enabled != command.equals(OnOffType.ON)) {
+                                requiresUpdate = true;
+                            }
+                            curfew.enabled = (command.equals(OnOffType.ON));
+                        }
+                        break;
+                    case DEVICE_CHANNEL_CURFEW_LOCK_TIME:
+                        if ((command instanceof StringType) && (command.toString().matches(CURFEW_TIME_REGEXP))) {
+                            logger.debug("old locktime: {}, new locktime: {}", curfew.lockTime, command);
+
+                            if (!(curfew.lockTime.equals(command.toString())) && curfew.enabled) {
+                                requiresUpdate = true;
+                            }
+                            curfew.lockTime = command.toString();
+                        } else {
+                            logger.warn("Incorrect curfew time format HH:mm: {}", command);
+                        }
+                        break;
+                    case DEVICE_CHANNEL_CURFEW_UNLOCK_TIME:
+                        if ((command instanceof StringType) && (command.toString().matches(CURFEW_TIME_REGEXP))) {
+                            if (!(curfew.unlockTime.equals(command.toString())) && curfew.enabled) {
+                                requiresUpdate = true;
+                            }
+                            curfew.unlockTime = command.toString();
+                        } else {
+                            logger.warn("Incorrect curfew time format HH:mm: {}", command);
+                        }
+                        break;
+                    default:
+                        break;
+                }
+                logger.debug("requiresUpdate: {}", requiresUpdate);
```

Comment:
Can you either remove this or make it more meaningfully, now it looks like some remains of debugging code.

You could, for example, add inside the if a log that you are about to start setting the curfews.